### PR TITLE
Working with mu clone.

### PR DESCRIPTION
**Pull request by fabioz**

### CODE REVIEW

File: mu_repo/action_clone.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    https://github.com/myuser/projectA.git
+
+    So, the url: https://github.com/myuser or git@github.com:myuser has to be configured as the
+    'base' url for clones.
+
+    mu-repo in this case will work with git to provide that configuration, so, one should first
+    add that remote for mu-repo to work with:
+
+    git config --global --add mu-repo.remote-host ssh://git@github.com:myuser
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: I'm thinking `mu-repo.remote` would be more align with the other options used by git, as `remote` is the term used through the standard git config:

```ini
[remote "origin"]
	url = https://bruno@eden.esss.com.br/stash/scm/esss/etk.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "fb-dist-12.0.18"]
	remote = origin
	merge = refs/heads/fb-dist-12.0.18
```

File: mu_repo/action_clone.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    Note that it's possible to add as many urls as wanted (they'll all be checked later on
+    when cloning the project). Note that multiple remote hosts may be added that way.
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: The line with the `Note` is redundant. I think it would be better to get rid of it and mention that the remotes will be tried in the order they were added.

File: mu_repo/action_clone.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            repos.append(arg)
+        else:
+            other_cmd_line_args.append(arg)
+
+
+    git = params.config.git
+
+    if not repos:
+        # We do not have any base: do a regular clone
+        ExecuteCommand([git] + params.args)
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: Hmmm not sure if it wouldn't be more helpful instead to display a message saying that no `mu-repo.remote-host` is configured, and even providing an example on how to do so. 

But I also understand that in general the approach is to just forward subcommands to git... this makes me wonder if perhaps this should be another command, `mclone` perhaps?

File: mu_repo/action_clone.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        return
+
+    remote_hosts = params.config.remote_hosts
+
+    if not remote_hosts:
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: Hmmm not sure if it wouldn't be more helpful instead to display a message saying that no mu-repo.remote-host is configured, and even providing an example on how to do so.

But I also understand that in general the approach is to just forward subcommands to git... this makes me wonder if perhaps this should be another command, mclone perhaps?

File: mu_repo/action_clone.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                # Ok, it exists, let's grab the dependencies (if there are any)
+                created_dir = os.path.join('.', repo)
+                new_mu_repo = os.path.join(created_dir, '.mu_repo')
+                if os.path.exists(new_mu_repo):
+
+                    with open(os.path.join(new_mu_repo), 'r') as stream:
+                        mu_config = mu_repo.Config.Create(stream.read())
+
+                    for new_repo in mu_config.repos:
+                        if new_repo.startswith('../'):
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: I think it is safe to check for `startswith('..')`, as windows users may have configured `..\` and don't care about Windows.

I think that the fact that a repository in `mu_repo` must be relative in order for the "recursive clone" to work should be mentioned in the README, as otherwise it will simply be ignored and the user will be left wondering why things are not working.

File: mu_repo/action_clone.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                            # I.e.: can only clone repositories that would appear alongside
+                            # the repository just cloned.
+                            new_repo_name = new_repo[3:]
```
Comment (nicoddemus) [inline mu_repo/action_clone.py]: I know it is a stretch of a feature, but any reason why `mu` couldn't support `repoA=../../libA` setups?

File: mu_repo/action_clone.py (hunk 1 above)
Comment (fabioz) [inline mu_repo/action_clone.py]: I thought about that, but I think that remote already has the meaning of the remote repository, whereas mu needs to work with the host to then decide on the remote repository (which is why I called it remote-host)

File: mu_repo/action_clone.py (hunk 4 above)
Comment (fabioz) [inline mu_repo/action_clone.py]: Done. I think that as cloning doesn't make sense as it should not work on multiple projects anyway having a mu clone is Ok.

File: mu_repo/action_clone.py (hunk 2 above)
Comment (fabioz) [inline mu_repo/action_clone.py]: Done

File: mu_repo/action_clone.py (hunk 5 above)
Comment (fabioz) [inline mu_repo/action_clone.py]: Checking for ../ and ..\ now (and adding a print if skipped). Will add that to the docs.

File: mu_repo/action_clone.py (hunk 6 above)
Comment (fabioz) [inline mu_repo/action_clone.py]: Not really, just needs to be better tested and checked (as it has more corner cases depending on where the user is checking out)... Also, I'm not sure how valid this use-case is (although if there is such a use case, it's definitely doable).

File: mu_repo/action_clone.py (hunk 1 above)
Comment (nicoddemus) [inline mu_repo/action_clone.py]: Good point, thanks! :smile:

Comment (fabioz) [inline mu_repo/action_clone.py]: I think remote-base-url seems better (will change).

**Comment by nicoddemus** [comment]
Looks good! :+1: :smile:
